metrics-operator/controllers/common/providers/dynatrace/common.go

```diff
@@ -24,20 +25,37 @@ type Error struct {
 	Message string `json:"message"`
 }
 
-func getDTSecret(ctx context.Context, provider metricsapi.KeptnMetricsProvider, k8sClient client.Client) (string, error) {
+type SecretValues struct {
+	Token   string `json:"token"`
+	AuthUrl string `json:"authurl"`
+}
+
+func getDTSecret(ctx context.Context, provider metricsapi.KeptnMetricsProvider, k8sClient client.Client) ([]byte, error) {
 	if !provider.HasSecretDefined() || !provider.HasSecretKeyDefined() {
-		return "", ErrSecretKeyRefNotDefined
+		return []byte{}, ErrSecretKeyRefNotDefined
 	}
 	dtCredsSecret := &corev1.Secret{}
 	if err := k8sClient.Get(ctx, types.NamespacedName{Name: provider.Spec.SecretKeyRef.Name, Namespace: provider.Namespace}, dtCredsSecret); err != nil {
-		return "", err
+		return []byte{}, err
 	}
 
 	token := string(dtCredsSecret.Data[provider.Spec.SecretKeyRef.Key])
+	secretValues := SecretValues{
+		Token:   token,
+		AuthUrl: fmt.Sprintf("https://dev.%s.internal.dynatracelabs.com/sso/oauth2/%s", token, token),
```

Comment:
This is hardcoded, I guess we can only pass here, what is stored in the secret by the user

Reply:
And abt it, I am a bit confused 😕. Does the user himself provide the url which needs to be defined here ??

Reply:
the url is hardcoded here, I would say lines 44-45 are not needed here. Also this function returns the secret as []bytes, is there a reason for it? I would say it should return token and url in a format, that can be directly used further.